Modules/Core/Common/wrapping/test/itkPointSetTest.py

```diff
@@ -0,0 +1,52 @@
+# ==========================================================================
+#
+#   Copyright NumFOCUS
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#          https://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ==========================================================================*/
+
+import itk
+import unittest
+import numpy as np
+
+class PointSetTestCase(unittest.TestCase):
+    """Tests itk.PointSet"""
+
+    def test_SetPointsByCoordinates_from_NumPy_array(self) -> None:
+        """Tests the SetPointsByCoordinates method, having a NumPy array as input"""
+
+        # The pixel type is irrelevant for this test, just use unsigned char.
+        PixelType = itk.UC
+
+        # Test the most common dimensions, and a few different numbers as number of points.
+        for dimension in [2, 3]:
+            for number_of_points in [1, 4, 9]:
+                # Create a NumPy array of random coordinates between -1.0 and 1.0
+                np_array = 2.0 * np.random.rand(number_of_points * dimension) - 1.0
+
+                point_set = itk.PointSet[PixelType, dimension].New()
+                point_set.SetPointsByCoordinates(np_array)
+
+                points = point_set.GetPoints()
+
+                self.assertEqual(points.Size(), number_of_points)
+
+                for i in range(number_of_points):
+                    point = points.GetElement(i)
+                    for j in range(dimension):
+                        self.assertAlmostEqual(point[j], np_array[i * dimension + j])
```

Comment:
It is better to use [`numpy` `assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose), which is must faster and has a shape check.

Reply:
Thanks for the suggestion @thewtex Can you possibly type out exactly how you want it to be?

Reply:
@thewtex Also, do you think the test should be faster than it is right now?

Reply:
Maybe `np.assert_allclose(points, np_array.T)`? Does that compile and work?

Reply:
@dzenanz Thanks for the suggestion, but no, unfortunately:
```python
np.assert_allclose(points, np_array.T)
```
> module 'numpy' has no attribute 'assert_allclose'
----

```python
np.testing.assert_allclose(points, np_array)
```
> TypeError: unsupported operand type(s) for -: 'itkVectorContainerULLPF2' and 'float'
----

```python
np.testing.assert_allclose(point_set, np_array)
```
> TypeError: unsupported operand type(s) for -: 'itkPointSetUC2' and 'float'

Reply:
Maybe not for the whole array, but for a single point? And iterate over points.

Reply:
> Maybe not for the whole array, but for a single point?

Maybe... but I don't know how. So it would be helpful if Matt (@thewtex) could tell us, exactly. Otherwise we just keep guessing 🤷

Reply:
@N-Dekker 

```python
from numpy.testing import assert_allclose

pointset_dict = itk.dict_from_pointset(point_set)
assert_allclose(pointset_dict['points'], np_array))
```

Reply:
Thanks @thewtex, but sorry, that does not work for me either. See below. Did you locally try this particular unit test with your suggested adjustment? Would it be acceptable to you to just merge the pull request as it is right now, and then afterwards possibly make a new PR to further optimize the test, for example by using `numpy.testing.assert_allclose`?

```
ERROR: test_SetPointsByCoordinates_from_NumPy_array (__main__.PointSetTestCase.test_SetPointsByCoordinates_from_NumPy_array)
Tests the SetPointsByCoordinates method, having a NumPy array as input
----------------------------------------------------------------------
Traceback (most recent call last):
  File "itkPointSetTest.py", line 46, in test_SetPointsByCoordinates_from_NumPy_array
    pointset_dict = itk.dict_from_pointset(point_set)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cedekker\.conda\envs\ITKSwigPy311\Lib\site-packages\itk\support\extras.py", line 957, in dict_from_pointset
    pixel_type, mangle, pixel_type_components = wasm_type_from_pointset_type(pointset)
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cedekker\.conda\envs\ITKSwigPy311\Lib\site-packages\itk\support\helpers.py", line 393, in wasm_type_from_pointset_type
    return pixelType, python_to_js(mangle), pixel_type_components
                      ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cedekker\.conda\envs\ITKSwigPy311\Lib\site-packages\itk\support\helpers.py", line 212, in python_to_js
    return _python_to_js[name]
           ~~~~~~~~~~~~~^^^^^^
KeyError: None
```

Reply:
> Thanks @thewtex, but sorry, that does not work for me either. See below. Did you locally try this particular unit test with your suggested adjustment? 

No

> Would it be acceptable to you to just merge the pull request as it is right now, and then afterwards possibly make a new PR to further optimize the test, for example by using numpy.testing.assert_allclose?

No, this is an important performance and usage fix, and it is also important to verify that the proposed change is correct in terms of `shape`.

I will investigate locally.

Reply:
> it is also important to verify that the proposed change is correct in terms of `shape`.

@thewtex Regarding the `shape`, please note that it is important that the argument of `SetPointsByCoordinates` is a "flat" (1-D) array of coordinate values, when doing

```python
point_set.SetPointsByCoordinates(np_array)
```

The `SetPointsByCoordinates` member function is meant to be an alternative to the existing unsafe `SetPoints` overload  (issue #4848), which also expects a "flat" array of coordinate values as argument.

So the PointSet and the array of coordinates (`np_array`) have different shapes, but that is _intentionally_. So I'm sorry, I don't think `numpy.testing.assert_allclose` can be used for this unit test. 🤷

Reply:
It might, with reshaping. Reshaping in numpy is easy.

Reply:
I'm sorry, but this is just a single unit test. It is not meant to be a tutorial on how to reshape a numpy array.

Reply:
> The SetPointsByCoordinates member function is meant to be an alternative to the existing unsafe SetPoints overload (issue #4848), which also expects a "flat" array of coordinate values as argument.

> So the PointSet and the array of coordinates (np_array) have different shapes, but that is intentionally.

The intension and/or the related support functions may need improvements :-). NumPy arrays representing points should not be flat -- they should have a dimension that corresponds to the spatial dimension.